rcl_yaml_param_parser/src/parser.c

```diff
@@ -1105,7 +1106,27 @@ static void * get_value(
   {
     errno = 0;
     endptr = NULL;
-    dval = strtod(value, &endptr);
+    const char * iter_ptr = NULL;
+    if ((0 == strcmp(value, ".nan")) ||
+      (0 == strcmp(value, ".NaN")) ||
+      (0 == strcmp(value, ".NAN")) ||
+      (0 == strcmp(value, ".inf")) ||
+      (0 == strcmp(value, ".Inf")) ||
+      (0 == strcmp(value, ".INF")) ||
+      (0 == strcmp(value, "+.inf")) ||
+      (0 == strcmp(value, "+.Inf")) ||
+      (0 == strcmp(value, "+.INF")) ||
+      (0 == strcmp(value, "-.inf")) ||
+      (0 == strcmp(value, "-.Inf")) ||
+      (0 == strcmp(value, "-.INF")))
+    {
+      for (iter_ptr = value; !isalpha(*iter_ptr); ) {
+        iter_ptr += 1;
+      }
+      dval = strtod(iter_ptr, &endptr);
```

Comment:
I think we should set a minus flag if it exists(-.inf, -.Inf and -.INF), and then set `dval=-dval` if the flag is true.
In my opinion, "-.inf" is different from "+.inf". (at least, ros2 param shows the different value, such as '-inf' and 'inf'.)